src/context.js

```diff
@@ -0,0 +1,71 @@
+import React, { useState, useContext, useEffect, useCallback } from 'react';
+
+const url = 'https://www.thecocktaildb.com/api/json/v1/1/search.php?s=';
+const AppContext = React.createContext();
+
+const AppProvider = ({ children }) => {
+  const [loading, setLoading] = useState(true);
+  const [searchTerm, setSearchTerm] = useState('a');
+  const [cocktails, setCocktails] = useState([]); 
+  const [currentPage, setCurrentPage] = useState(1);
+  const [cocktailsPerPage] = useState(6); 
+
+//get current cocktail
+  const indexOfLastCocktail = currentPage * cocktailsPerPage; 
+  const indexOfFirstCocktail = indexOfLastCocktail - cocktailsPerPage; 
+  const currentCocktail = cocktails.slice(indexOfFirstCocktail, indexOfLastCocktail);
+
+  const paginate = pageNumber => setCurrentPage(pageNumber);
+
+  const fetchDrinks = useCallback( async () => {
+    setLoading(true)
+    try {
+      const response = await fetch(`${url}${searchTerm}`)
+      const data = await response.json()
+      console.log(data);
+      const { drinks } = data;
+      if (drinks) {
+        const newCocktails = drinks.map((item) => {
+          const {
+            idDrink,
+            strDrink,
+            strDrinkThumb,
+            strAlcoholic,
+            strGlass,
+          } = item
+
+          return {
+            id: idDrink,
+            name: strDrink,
+            image: strDrinkThumb,
+            info: strAlcoholic,
+            glass: strGlass,
+          }
+        })
+        setCocktails(newCocktails)
+      } else {
+        setCocktails([])
+      }
+      setLoading(false)
+    } catch (error) {
+      console.log(error)
+      setLoading(false)
+    }
+  },[searchTerm])
+  useEffect(() => {
+    fetchDrinks()
+  }, [searchTerm,fetchDrinks])
+  return (
+    <AppContext.Provider
```

Comment:
Is there any reason why you decided to use context for your app?

Reply:
I thought it would be easier to hold the data in component. Do you think I should do another method?

Reply:
Well yeah but if you will be working on more data it will make this file really huge and complex. I think it's better to keep the logic close to the component that is using it. You can pretty much do without this file by just using state and props in the `cocktailList` and `cocktail` components. Also, this approach will also make automated testing a bit tricky. I know it works but sometimes simple is better

Reply:
Ok perfect. I will make that change as well.